src/Elastic.Apm.SmartSql/SmartSqlDiagnosticListener.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Data;
+using Elastic.Apm.Api;
+using Elastic.Apm.DiagnosticSource;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Model;
+using SmartSql.Diagnostics;
+
+namespace Elastic.Apm.SmartSql
+{
+	internal class SmartSqlDiagnosticListener : IDiagnosticListener
+	{
+		private readonly ConcurrentDictionary<Guid, Span> _spans = new ConcurrentDictionary<Guid, Span>();
+
+		public SmartSqlDiagnosticListener(IApmAgent agent) => Logger = agent.Logger?.Scoped(nameof(SmartSqlDiagnosticListener));
+
+		private ScopedLogger Logger { get; }
+
+		public string Name => "SmartSqlDiagnosticListener";
+
+		public void OnCompleted() { }
+
+		public void OnError(Exception error) { }
+
+		public void OnNext(KeyValuePair<string, object> kv)
+		{
+			switch (kv.Key)
+			{
+				case string k when k == SmartSqlDiagnosticListenerExtensions.SMART_SQL_BEFORE_COMMAND_EXECUTER_EXECUTE:
+					if (kv.Value is CommandExecuterExecuteBeforeEventData executerExecuteBeforeEventData)
+					{
+						var newsSpan = Agent.TransactionContainer.Transactions.Value.StartSpanInternal(executerExecuteBeforeEventData.Operation,
```

Comment:
The current transaction can be null. This is when there is no transaction started. In the sample application this never happens, because for each incoming HTTP request our ASP.NET Core auto instrumentation starts a transaction. 

I think this one can help you - it should be similar: https://github.com/elastic/apm-agent-dotnet/blob/master/src/Elastic.Apm.EntityFrameworkCore/EfCoreDiagnosticListener.cs#L31 - spot `&& Agent.TransactionContainer.Transactions.Value != null`

Reply:
ok,I will add null check in next pr.